oauth-common/src/main/java/io/strimzi/kafka/oauth/metrics/AbstractMetricKeyProducer.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017-2022, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.metrics;
+
+import java.net.URI;
+
+public abstract class AbstractMetricKeyProducer implements MetricKeyProducer {
```

Comment:
This abstraction seems really weak. Is there really much value in a class holding two fields like this, or could they just be redeclared in the subclasses?